AntonioLok/online-shopping-reactJS-Backend
# add login route

# Changelog
Adding login route. For this, I am using bcryptjs to validate password being passed from the client against the hashed one in the database. I am also using the package `jsonwebtoken` to sign the given payload (userId) into a JSON Web Token string that is send as a response to the client. 

# Test
- Pull latest changes `git pull`
- Install latest dep `npm i`
- Start the server `SECRET_KEY=SECRET npm run start`. Note that I've set the secret key in my environment variables as I did not want to expose the secret key in my code.
- Create a new user by sending a POST request to `http://localhost:8000/users/register`as shown below:
![image](https://user-images.githubusercontent.com/14690224/59982603-f78ef700-95e2-11e9-8f6b-cbcf6124307d.png)
- Now send another POST request using the same body (same id and password as the acc you just registered) to `http://localhost:8000/users/login`. You should get a data payload with the token generated from the server as shown below.
![image](https://user-images.githubusercontent.com/14690224/59982718-7cc6db80-95e4-11e9-89dd-e6f7ecd0d50c.png)
